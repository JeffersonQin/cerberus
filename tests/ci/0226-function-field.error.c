typedef void fun (void);
struct S {
  fun f;
};
