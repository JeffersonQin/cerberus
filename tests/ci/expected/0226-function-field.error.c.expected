ci/0226-function-field.error.c:3:7: error: constraint violation: member 'f' declared as a function
  fun f;
      ^ 
ยง6.7.2.1#3, sentence 1: 
3   A structure or union shall not contain a member with incomplete or function type (hence,
    a structure shall not contain an instance of itself, but may contain a pointer to an instance
    of itself), except that the last member of a structure with more than one named member
    may have incomplete array type; such a structure (and any union containing, possibly
    recursively, a member that is such a structure) shall not be a member of a structure or an
    element of an array.
